ctapipe/reco/sklearn.py

```diff
@@ -507,6 +512,353 @@ def predict_table(self, key, table: Table) -> Table:
         return result
 
 
+class DispReconstructor(Reconstructor):
+    """
+    Predict absolute value and sign for disp origin reconstruction for each telescope.
+    """
+
+    property = "geometry"
+    prefix = Unicode(default_value="disp", allow_none=False).tag(config=True)
+    features = List(Unicode(), help="Features to use for both models").tag(config=True)
+
+    norm_target = "true_norm"
+    norm_config = Dict({}, help="kwargs for the sklearn regressor").tag(config=True)
+    norm_cls = Enum(
+        SUPPORTED_REGRESSORS.keys(),
+        default_value=None,
+        allow_none=True,
+        help="Which scikit-learn regression model to use.",
+    ).tag(config=True)
+
+    norm_log_target = Bool(
+        default_value=False,
+        help="If True, the norm model is trained to predict the natural logarithm.",
+    ).tag(config=True)
+
+    sign_target = "true_sign"
+    sign_config = Dict({}, help="kwargs for the sklearn classifier").tag(config=True)
+    sign_cls = Enum(
+        SUPPORTED_CLASSIFIERS.keys(),
+        default_value=None,
+        allow_none=True,
+        help="Which scikit-learn classification model to use.",
+    ).tag(config=True)
+
+    sign_invalid_class = Integer(
+        default_value=0,
+        help="The label to fill in case no sign prediction could be made",
+    ).tag(config=True)
+
+    stereo_combiner_cls = ComponentName(
+        StereoCombiner,
+        default_value="StereoMeanCombiner",
+        help="Which stereo combination method to use",
+    ).tag(config=True)
+
+    load_path = Path(
+        default_value=None,
+        allow_none=True,
+        help="If given, load serialized model from this path",
+    ).tag(config=True)
+
+    def __init__(self, subarray, norm_models=None, sign_models=None, **kwargs):
+        # Run the Component __init__ first to handle the configuration
+        # and make `self.load_path` available
+        Component.__init__(self, **kwargs)
+
+        if self.load_path is None:
+            if self.norm_cls is None or self.sign_cls is None:
+                raise TraitError(
+                    "Must provide `norm_cls` and `sign_cls` if not loading form file"
+                )
+
+            if subarray is None:
```

Comment:
How can this happen? subarray is a posarg in `__init__`

Reply:
Oh I missed that. In 'SKLearnReconstrctor´ it is 
```python
def __init__(self, subarray=None, models=None, **kwargs):
            
            ...
    if subarray is None:
        raise TypeError(
            "__init__() missing 1 required positional argument: 'subarray'"
        )
```
So the error message is a bit misleading since `subarray` isn't a posarg there. Would it be better to always make it a posarg and remove this error message since it is unnecessary?

Reply:
I think the `None` case is needed for `.read()` ing the pickled class on its own.

Reply:
Ok. Then I will do it the same way in `DispReconstructor`